File: BinaryTree/php/treeSum.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function treeSum(?TreeNode $root): int
{
    if ($root === null) {
        return 0;
    }

    return $root->val + treeSum($root->left) + treeSum($root->right);
}

File: BinaryTree/java/treeSum.java
class TreeSum {
  public static int treeSum(TreeNode root) {
    if (root == null) {
      return 0;
    }
    return root.val + treeSum(root.left) + treeSum(root.right);
  }

  public static void run() {
    // intentionally empty
  }
}
